consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -181,10 +183,17 @@ impl<T: Payload> RecoveryData<T> {
     fn find_root(
         blocks: &mut Vec<Block<T>>,
         quorum_certs: &mut Vec<QuorumCert>,
-        root_from_storage: HashValue,
+        storage_ledger: &LedgerInfo,
     ) -> Result<(Block<T>, QuorumCert, QuorumCert)> {
+        let root_from_storage = storage_ledger.consensus_block_id();
+        info!(
+            "The last committed block id as recorded in storage: {}",
+            root_from_storage
+        );
+
         // sort by round to guarantee the topological order of parent <- child
         blocks.sort_by_key(Block::round);
+        quorum_certs.sort_by_key(QuorumCert::certified_block_round);
```

Comment:
Curious why this is here?  It doesn't appear to be necessary, or even improve performance as we are searching for the max round of the qc (which will be at the end?).  Also running the test without it passes.